BE/src/main/java/louie/dong/airbnb/accommodation/CustomAccommodationRepositoryImpl.java

```diff
@@ -0,0 +1,58 @@
+package louie.dong.airbnb.accommodation;
+
+import static louie.dong.airbnb.accommodation.QAccommodation.*;
+import static louie.dong.airbnb.accommodation.QAccommodationImage.*;
+import static louie.dong.airbnb.book.QBook.*;
+import static louie.dong.airbnb.wishlist.QWish.*;
+
+import com.querydsl.core.types.Predicate;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import louie.dong.airbnb.accommodation.dto.AccommodationSearchRequest;
+import louie.dong.airbnb.wishlist.QWish;
+
+@RequiredArgsConstructor
+public class CustomAccommodationRepositoryImpl implements CustomAccommodationRepository {
+
+    private final JPAQueryFactory queryFactory;
+
+    @Override
+    public List<Accommodation> searchAccommodations(
+        AccommodationSearchRequest accommodationSearchRequest) {
+        return queryFactory
+            .selectFrom(accommodation).distinct()
+            .leftJoin(accommodation.accommodationImages, accommodationImage)
+                .on(accommodationImage.imageType.eq(ImageType.MAIN))
+            .leftJoin(accommodation.wishlist).fetchJoin()
+            .leftJoin(accommodation.books, book).on(book.isCanceled.eq(false))
+            .where(
+                accommodation.country.contains(accommodationSearchRequest.getCountry()),
+                checkPrice(accommodationSearchRequest.getMinPrice(),
+                    accommodationSearchRequest.getMaxPrice()),
+                checkInAndOut(accommodationSearchRequest),
+                checkGuestCount(accommodationSearchRequest.getGuestCount())
+            ).fetch();
+    }
+
+    private BooleanExpression checkGuestCount(Integer guestCount) {
+        return guestCount != null ? accommodation.roomInformation.maxGuestCount.goe(guestCount) : null;
+    }
```

Comment:
삼항 연산자는 호불호(가독성)가 많이 갈려서 크게 추천하지는 않습니다.

```suggestion
    private BooleanExpression checkGuestCount(Integer guestCount) {
        if(guestCount != null){
            accommodation.roomInformation.maxGuestCount.goe(guestCount);
        }
        return null;
    }
```

Reply:
제가 쿼리 dsl은 사용 안해서 그런데 guestCount가 null일 경우 오류가 발생하나요?
아니면 null인 경우 실제 실행되는 쿼리가 달라지는지 궁금하네요

Reply:
Querydsl에서 사용하는 where 메서드의 파라메터에 null이 입력된다면 오류가 발생하지 않고 아래 사진처럼 해당 부분에 대한 쿼리를 제외하게 됩니다.
해당 기능을 통해서 동적 쿼리를 쉽게 구현할 수 있습니다. 😀
![image](https://user-images.githubusercontent.com/92966772/173222015-2cfc2738-201a-4941-af45-1833e95d58da.png)